dockstore-webservice/src/test/java/core/WDLParseTest.java

```diff
@@ -130,6 +134,49 @@ private static SourceFile getSourceFile2() {
     }
 
 
+    /**
+     * Tests that Dockstore can handle a workflow with locally recursive imports
+     */
+    @Test
+    public void testLocallyRecursiveImport() {
+        String type = "workflow";
+        File recursiveWDL = new File(ResourceHelpers.resourceFilePath("local-recursive-import/localrecursive.wdl"));
+        //String primaryDescriptorFilePath = recursiveWDL.getAbsolutePath();
+        String primaryDescriptorFilePath = "localrecursive.wdl";
+        SourceFile sourceFile = new SourceFile();
+
+        File recursiveimportWDL = new File(ResourceHelpers.resourceFilePath("local-recursive-import/first-import.wdl"));
+        SourceFile importsourceFile = new SourceFile();
+
+        try {
+            sourceFile.setContent(FileUtils.readFileToString(recursiveWDL, StandardCharsets.UTF_8));
+            sourceFile.setAbsolutePath("/localrecursive.wdl");
+            sourceFile.setPath("localrecursive.wdl");
+            sourceFile.setType(DescriptorLanguage.FileType.DOCKSTORE_WDL);
+
+            importsourceFile.setContent(FileUtils.readFileToString(recursiveimportWDL, StandardCharsets.UTF_8));
+            importsourceFile.setAbsolutePath("/first-import.wdl");
+            importsourceFile.setPath("first-import.wdl");
+            importsourceFile.setType(DescriptorLanguage.FileType.DOCKSTORE_WDL);
+
+            Set<SourceFile> sourceFileSet = new HashSet<>();
+            sourceFileSet.add(sourceFile);
+
+            sourceFileSet.add(importsourceFile);
+
+            WDLHandler wdlHandler = new WDLHandler();
+            VersionTypeValidation validation = wdlHandler.validateEntrySet(sourceFileSet, primaryDescriptorFilePath, type);
+            Assert.assertFalse(validation.isValid());
+            // Go through message part and verify it says 'Recursive local import detected'
+            Optional<Map.Entry<String, String>> validationEntryMap = validation.getMessage().entrySet().stream()
```

Comment:
If instead of `validation.getMessage().entrySet...` you did `validation.getMessage.values()...` and it would make your code a little less complicated.

Reply:
fixed